File: src/Learn/Arrays101/CheckDoubleNExist.cs
public class Solution {
    public bool CheckIfExist(int[] arr) {
        HashSet<int> lookup = new HashSet<int>();
        foreach(int num in arr){
        	if((num % 2 == 0 && lookup.Contains(num / 2)) || lookup.Contains(2 * num)) return true;
        	else lookup.Add(num);
        }
        return false;
    }
}

File: src/Learn/Arrays101/checkDoubleNExist.cpp
class Solution {
public:
    bool checkIfExist(vector<int>& arr) {
        int n = arr.size();
        for(int i = 0; i < n - 1; i++){
        	for(int j = i + 1; j < n; j++){
        		if(2 * arr[i] == arr[j] || 2 * arr[j] == arr[i]) return true;
        	}
        }
        return false;
    }
};